src/platform/PICO/system.c

```diff
@@ -85,9 +86,15 @@ void systemReset(void)
     //TODO: implement
 }
 
+uint32_t *systemUniqueId = NULL;
+
 void systemInit(void)
 {
     //TODO: implement
+
+    pico_unique_board_id_t id;
+    pico_get_unique_board_id(&id);
+    systemUniqueId = (uint32_t*)id.id;
```

Comment:
If you insist keeping UID0 == 0, but maybe fill lower 2 words,
```suggestion
    memcpy(&systemUniqueId + 1, &id.id, 8); 
```

or, storing from index 0:
```suggestion
    memcpy(&systemUniqueId, &id.id, MIN(sizeof(systemUniqueId), PICO_UNIQUE_BOARD_ID_SIZE_BYTES)); 
```

Reply:
The reason I opted for the pointer is the SDK already stores a memory representation. This method will duplicate the value - albeit minor and of no real concern.